libpod/runtime_img.go

```diff
@@ -76,6 +80,79 @@ type ImageFilterParams struct {
 	ImageInput       string
 }
 
+// struct for when a user passes a short or incomplete
+// image name
+type imagePullStruct struct {
+	imageName   string
+	tag         string
+	registry    string
+	hasRegistry bool
+	transport   string
+}
+
+func (ips imagePullStruct) returnFQName() string {
+	return fmt.Sprintf("%s%s/%s:%s", ips.transport, ips.registry, ips.imageName, ips.tag)
+}
+
+func getRegistriesToTry(image string) ([]string, error) {
+	var registries []string
+	var imageError = fmt.Sprintf("unable to parse '%s'\n", image)
+	imgRef, err := reference.Parse(image)
+	if err != nil {
+		return nil, errors.Wrapf(err, imageError)
+	}
+	tagged, isTagged := imgRef.(reference.NamedTagged)
```

Comment:
(Note that imgRef may also be [_host_/]_repo_`@`_algorithm_`:`_digest_. See below.)

Reply:
@baude has refactored the `registriesToTry` code a lot, which he will push with the kpod create PR.
@mtrmac do you still want me to make these changes or should we get this in and then make it when the new code gets in?

Reply:
\*shrug\* I guess it wouldn’t be fair to block this PR on things it is not introducing. (Being lazy, I’d love for my comments not to get lost here; but I can be less lazy.)

From my POV, _this_ PR does need to either decide whether `images` should be processed completely or only until first success — or not touch the `tryRegistries`/`images` code and let this be solved somewhere else.